examples/blame_example.ipynb

```diff
@@ -90,87 +90,97 @@
     "    for t, agent_states in enumerate(agent_states_history):\n",
-    "        num_agents = len(agent_states)\n",
-    "        if not num_agents == len(agent_attributes):\n",
+    "        if not len(agent_states) == len(agent_attributes):\n",
     "            raise Exception(\"Incorrect number of agents or agent attributes.\")\n",
     "\n",
-    "        #Transform the points of the agents polygon into the common world coordinate frame\n",
-    "        polygons = [None]*num_agents\n",
-    "        for i, (state, attributes) in enumerate(zip(agent_states,agent_attributes)):\n",
-    "            dx = attributes.length/2\n",
-    "            dy = attributes.width/2\n",
-    "\n",
-    "            origin = np.array([state.center.x, state.center.y])\n",
-    "\n",
-    "            angle = state.orientation\n",
-    "            c, s = np.cos(angle), np.sin(angle)\n",
-    "            \n",
-    "            #Rotation matrix from agent coordinate frame to world coordinate frame\n",
-    "            R = np.array([[c, s],\n",
-    "                          [-s, c]])\n",
-    "            #Stacked vectors of agent polygon points in the agents coordinate frame\n",
-    "            #Points are in order around the polygon\n",
-    "            V = np.array([[dx,dy],[dx,-dy],[-dx,-dy],[-dx,dy]])\n",
-    "            #Apply the rotation into the world frame\n",
-    "            VR = np.matmul(V,R)\n",
-    "            \n",
-    "            #Add the displacement between the world frame and the agent frame\n",
-    "            polygons[i] = origin + VR\n",
-    "\n",
-    "        #Check every combination of of agent polygons\n",
-    "        detected_collisions = []\n",
-    "        polygons = [Polygon([p[0], p[1], p[2], p[3]]) for p in polygons]\n",
-    "        for j in range(num_agents):\n",
-    "            for k in range(j+1,num_agents):\n",
-    "                try:\n",
-    "                    if polygons[j].intersection(polygons[k]).area:\n",
-    "                        detected_collisions.append((j,k))\n",
-    "                except GEOSException as e:\n",
-    "                    print(f\"Collision candidates {j} and {k} failed with error {e}.\")\n",
-    "                    pass\n",
-    "        \n",
-    "        #Check for collisions and acquire: agent IDs, collision start time, collision end time\n",
-    "        #Collect all collisions in the simulation\n",
-    "        for agent_tuple in detected_collisions:\n",
-    "            #If a colliding pair is detected and are not in the dictionary, add them here to track them\n",
+    "        polygons = transform_all_agent_vertices_into_world_frame(agent_states,agent_attributes)\n",
+    "        detected_agent_pairs = check_agent_pairwise_intersections(polygons)    \n",
+    "              \n",
+    "        for agent_tuple in detected_agent_pairs:\n",
     "            if not agent_tuple in collisions_ongoing:\n",
```

Comment:
`if agent_tuple not in collisions_ongoing`

Reply:
Done